### ISSUE TITLE
device.clean() should not set reportedProp to null

### ISSUE THREAD

**Issue by carocad** [Azure/azure-iot-sdk-java#164]
OS and version used: MAC OS X 10.11.6

Java runtime used: Android JDK 1.7

SDK version used: iot-device-client:1.5.35

# Description of the issue:
The abstract class `Device` in `package com.microsoft.azure.sdk.iot.device.DeviceTwin;` contains a method `clean()` which sets the `reportedProp` to null. However if a user calls `setReportedProp` afterwards, the code throws a `NullPointerException` because `reportedProp` was set to `null` instead of being [clear](https://docs.oracle.com/javase/7/docs/api/java/util/HashSet.html#clear())

# Code sample exhibiting the issue:
```java
client = new DeviceClient(connString, protocol);
final Device homeKit = new Device() {
            @Override
            public void PropertyCall(String propertyKey, Object propertyValue, Object context) {
                Log.i(TAG, propertyKey + " changed to " + propertyValue);
                try {
                    this.setReportedProp(new Property(propertyKey, propertyValue));
                    client.sendReportedProperties(this.getReportedProp());
                    this.clean();
                    return;
                } catch (Exception e) {
                    // null pointer exception whenever a property is changed
                    e.printStackTrace();
                }
            }
        };
client.startDeviceTwin(new DeviceTwinStatusCallBack(), null, homeKit, null);
```

**Comment by timtay-microsoft** [comment]
Sorry for the delay on this fix. The fix has been checked in as of b1b4a462a58588e86af7441278e8e3453fae7218

The fix will be in the next release. I will update this thread when that happens

**Comment by timtay-microsoft** [comment]
It will take a few days for the release to propagate up to Maven central, but our latest release will fix this issue for you:

https://github.com/Azure/azure-iot-sdk-java/releases/tag/2018-1-23

I am closing this issue for now. Feel free to re-open if you still run into problems here.
